src/shogun/regression/LeastAngleRegression.cpp

```diff
@@ -102,37 +102,6 @@ void CLeastAngleRegression::plane_rot(ST x0, ST x1,
 	}
 }
 
-bool CLeastAngleRegression::train_machine(CFeatures* data)
-{
-	if (!data)
-	{
-		REQUIRE(features, "No features provided.\n")
-		REQUIRE(features->get_feature_class() == C_DENSE,
-			"Feature-class (%d) must be of type C_DENSE (%d)\n", features->get_feature_class(), C_DENSE)
-			
-		data = features;
-	}
-	else
-		REQUIRE(data->get_feature_class() == C_DENSE,
-			"Feature-class must be of type C_DENSE (%d)\n", data->get_feature_class(), C_DENSE)
```

Comment:
nice, so we shaved off all this code in every class that dispatches the templated type for training